### ISSUE TITLE
Cura accessibility

### ISSUE THREAD

**Issue by MikeZimmer25** [Ultimaker/Cura#4281]
Hello,

i am programming with cura 1.4 because the newer versions are not accessible for me. I can't se very good an work with a screenreader like JAWS or NVDA but i can see my ovjects in open scad. Now i search for the ironing function and find out, that in the newest version (3.2) of cure this function is includet. Like i wrote, i canÄt work with newer versions. Can someone create me a plugin for cura 1.4 or tell me, how i can create it self or find it somewhere? I would be happy, if the next version of cure is more accessible for visual improved persons.

Thank you very much

**Application Version**
 1.4

**Platform**
 Windows 7 64

**Printer**
 Ultimake 2 Extended

**Steps to Reproduce**

**Actual Results**

**Expected results**

**Additional Information**

**Comment by fieldOfView** [comment]
The ironing functionality (or any other new functionality) will not be backported to an older version of Cura, especially a version < 2.7.

Having said that, I am sorry to hear that the new Cura is less accessible than the old Cura. I have never used a screenreader myself, and I don't know what works and what does not work (or even what would be required for you to be able to use Cura). Can you elaborate a bit?

PS: it was probably a typo, but I like the "visually improved persons" bit.

**Comment by MikeZimmer25** [comment]
Hello,

one thing i forgot:

In Cura 1.4 the colors of the frontend ar the same as windows system colors. In the newer versions the colors are fixed, so the background color is light and the text colors ar light too. It were better, if the colors ar the same as the system colors too.

Thank you

Sahin

**Comment by fieldOfView** [comment]
The interface for Cura is developed with QML and pyqt.

QML has specific support for screenreaders, but it it currently not used in Cura. It probably should be.
http://doc.qt.io/qt-5/qml-qtquick-accessible.html

At least the sidebar in Cura should be fully tab-indexed (since juli last year). Does that work for you?

I'm guessing you use a high-contrast system palette, right? Cura uses a custom theme, but I can make a high-contrast Cura theme for you. Making a theme that automatically uses system colors would be better, but is currently less likely to happen soon.

**Comment by MikeZimmer25** [comment]
Hello,

a few day ago i tried cura 3.4. With tab i could step from field to field but the screenreader only read the values in the edit field but not the labels. If you create a high contrast frontend, i will be very happy. The best ist black background and white text and borders of the edit fields.

Thank you for your very fast answer.

Best reguards

Sahin

**Comment by MikeZimmer25** [comment]
Hello,

i clicked on the link for qml. Like in Visual Basic, in witch i wrote and write some programms, there are three attributes, accessibility name, rule and discription. They must specified for all editfields, listboxes and other elements. So, if you want, i can try help you.

Best reguards

Sahin

**Comment by MikeZimmer25** [comment]
Hello,

sorry for my very fast answer, but did you something for my problem?

Best reguards

Sahin

**Comment by MikeZimmer25** [comment]
Hello,

i closed this thread again, but i reopened ist. And i see, that i wrote "answer" instad "question".

**Comment by MikeZimmer25** [comment]
Hello,

can you purhaps describe me, how you release the ironing functionality. Then i can edit my gcode in the same way. I tried the folowing way:
1. I create a cube width cube([50,50,3.4])
2. In Cura i add "Tweak at" and set the flow rate to 10%
3. In gcode file i deleated the layer number 110
4. In gcode file i change the layer distance of the last layer to 3.00
5. I printed the model

It don't work. Today i want to trie to set the speed to 10% and set the nothle temperature to 260. I think that the higher temperature and the lower speed has mor time to heatup the top layer.

Best reguards

Sahin

**Comment by Ghostkeeper** [comment]
Ironing basically works like this:
* Take the top layer polygon (for every layer, subtract the areas that are also in the layer above it).
* Perform an inset on this with half a line width (by default).
* Perform a travel move to one corner of the ironed area.
* Fill it with the desired pattern at 100% density (zigzag pattern by default) and rotated 90 degrees from the direction of the skin on that layer.
* Use a flow rate of 10% (by default) and a line width of 0.1 (by default).
* Draw these lines 10 microns lower than the normal position of the layer.

You can sort of mimic this effect by copying the last layer, reducing the flow to 5% and moving the copied layer down 0.01mm. Then repeat this layer 2 or 3 times to mimic the effect of the 0.1mm line width. The real trick of this is the line width, not the flow rate.

**Comment by Ghostkeeper** [comment]
We're looking into the accessibility library for the next release. The original instigator of getting accessibility was funnily enough not from our users but from an automated testing perspective: It also allows a GUI testing library to find the elements to click on.

We've discussed the possibility of adding a high contrast theme today but we decided that we don't have time for it now. A PR is welcome though.

**Comment by MikeZimmer25** [comment]
Hello,

did i understood you message correct? I print the last layer 2 or 3 times 0.01mm under the laqst position? You wrote also that the line width must be 0.1 mm. If i change the layer width to 0.1 mm than there must be more lines? Sorry, but can you send me an example for e 5x5x5 mm cube? I can open the .gcode-File and read the points of the printhead.

Thank you

Best reguards

**Comment by Ghostkeeper** [comment]
It prints the last layer at a normal 0.1mm layer thickness. And then it printed the ironing 0.01mm "into" that. Pressing down the ironing on the wrinkly clothing as it were.

We've since removed the 0.01mm so it doesn't apply any more. It now puts the ironing on exactly the same height.

[test.zip](https://github.com/Ultimaker/Cura/files/2919014/test.zip)

> You wrote also that the line width must be 0.1 mm. If i change the layer width to 0.1 mm than there must be more lines?

Layer height and line width are two different things. If you reduce the line width of ironing, then there will indeed be more lines. If you change the layer height, the number of lines to iron will not change.

**Comment by MikeZimmer25** [comment]
Hello,
until now i tried some versions of Cura. Today i downloaded Cura 4.6.1. It crashes when i click on "Get started" button without an error message. I checked my directX Version and uninstall and reinstall my graphic card drivers. I hav an ATI Radeon HD 7970. Can you help me, cura make running?
Thanks

**Comment by Ghostkeeper** [comment]
Please create a new issue rather than posting in this unrelated one from 2 years ago. You'll be faced with an issue template then and if you fill that in we might be able to help out.
